recipes-app/iot2050-firmware-update/files/iot2050-firmware-update.py

```diff
@@ -0,0 +1,200 @@
+            FirmwareUpdateException("Upgrade is not supported for the board")
+
+    def update_firmware(self, filename):
+        """Update Firmware"""
+        mtd_num = 0
+        img_pos = 0
+
+        print("===================================================")
+        print("IOT2050 firmware update started - DO NOT INTERRUPT!")
+        print("===================================================")
+
+        while True:
+            mtd_sys_path = "/sys/bus/platform/devices/47040000.spi/mtd/mtd{}".format(mtd_num)
+            mtd_name_path = "{}/name".format(mtd_sys_path)
+            mtd_size_path = "{}/size".format(mtd_sys_path)
+            mtd_erasesize_path = "{}/erasesize".format(mtd_sys_path)
+            mtd_dev_path = "/dev/mtd{}".format(mtd_num)
+
+            mtd_size = self.get_path_type_value(mtd_size_path)
+            mtd_size = int(mtd_size)
+
+            mtd_erasesize = self.get_path_type_value(mtd_erasesize_path)
+            mtd_erasesize = int(mtd_erasesize)
+
+            mtd_name = self.get_path_type_value(mtd_name_path)
+
+            print("Updating %-20s" % mtd_name.strip(),end="")
+            if mtd_name.strip() == "ospi.rootfs":
+                print("\n\nCompleted. Please reboot the device\n")
+                sys.exit(0)
+
+            mtd_pos = 0
+            while mtd_pos < mtd_size:
+                try:
+                    with open(mtd_dev_path, 'rb') as mtd:
+                        mtd.seek(mtd_pos)
+                        mtd_contents = mtd.read(mtd_erasesize)
+                except FileNotFoundError:
+                    print("Open %s failed" % mtd_dev_path)
+                    sys.exit(1)
+
+                with open(filename, 'rb') as firmware:
+                    firmware.seek(img_pos)
+                    firmware_contents = firmware.read(mtd_erasesize)
+
+                if not mtd_contents == firmware_contents:
+                    print("U", end="")
+                    sys.stdout.flush()
+                    self.flash_erase(mtd_dev_path, mtd_pos, mtd_erasesize)
+                    with open(mtd_dev_path, 'wb') as mtd_unity:
+                        mtd_unity.seek(mtd_pos)
+                        mtd_unity.write(firmware_contents)
+                else:
+                    print(".", end="")
+                    sys.stdout.flush()
+                mtd_pos = mtd_pos+mtd_erasesize
+                img_pos = img_pos+mtd_erasesize
+            print("\n")
+            mtd_num = (mtd_num+1)
+
+
+def main(argv):
+    usage = "%prog [Options] Firmware"
+    parser = OptionParser(usage)
+    parser.add_option('-f', '--force', action='store_true', dest='force', default=False, help="Force update, ignoring image checksum or device ID mismatches")
+
+    (options, args) = parser.parse_args()
+
+    if not args:
+        print(parser.parse_args(['-h']))
+        sys.exit(1)
+
+    erase_env_input = input("\nWarning: Environment variables set by the user may be destroyed. Continue (y/N)?")
```

Comment:
I would propose a little bit tuning of the working:
`"\nWarning: All U-Boot environment variables will be reset to factory settings. Continue (y/N)? "`